File: manager-web/src/main/resources/static/js/service/specificationService.js
// service层
// 创建一个service，名字叫brandService
app.service('specificationService',function($http){
    // 增加
    this.add = entity=>{
        return $http.post('/specification/add',entity)
    }
    // 修改
    this.update = entity =>{
        return $http.post('/specification/modify',entity)
    }
    // 删除
    this.delete = ids => {
        return $http.post('/specification/delete',ids)
    }
    // 根据ID查询
    this.findById = id => {
        return $http.get(`/specification/${id}`)
    }
    // 分页查询所有
    this.findAll = (page,size,searchEntity)=>{
        return $http.post(`/specification/list?page=${page}&size=${size}`,searchEntity)
    }
})

File: sellergoods-service/src/main/java/com/zephon/sellergoods/service/SpecificationService.java
package com.zephon.sellergoods.service;

import com.github.pagehelper.PageInfo;
import com.zephon.pojo.TbSpecification;
import com.zephon.pojo.TbSpecificationOption;

import java.util.List;
import java.util.Map;

/**
 * @author Zephon
 * @version V1.0
 * @Package com.zephon.sellergoods.service
 * @date 2020/6/29 下午4:15
 * @Copyright ©
 */
public interface SpecificationService {
    PageInfo<TbSpecification> findAll(Integer page, Integer size, TbSpecification specification);

    int add(TbSpecification specification);

    TbSpecification findById(Long id);

    int updateById(TbSpecification specification);

    int deleteByIds(List<Long> ids);

    List<Map<String, Object>> selectOptionList();
}
